gymnasium/utils/passive_env_checker.py

```diff
@@ -265,6 +265,48 @@ def env_step_passive_checker(env, action):
     return result
 
 
+def _check_render_return(render_mode, render_return):
+    """Produces warning if `render_return` doesn't match `render_mode`."""
+    # This many if statements are ugly, but we need it for short-circuiting in the rgb_array case (dict approach become difficult here)
+    if render_mode == "human" and render_return is not None:
+        logger.warn(f"Human rendering should return `None`, got {type(render_return)}")
+    if render_mode == "rgb_array":
+        if not isinstance(render_return, np.ndarray):
+            logger.warn(
+                f"RGB-array rendering should return a numpy array, got {type(render_return)}"
+            )
+        if isinstance(render_return, np.ndarray) and render_return.dtype != np.uint8:
```

Comment:
I don't think that will really work. It might be that multiple warnings should be issued (e.g. if neither the shape, nor the dtype is correct), so we can't just do Elifs. We could put everything into an else block and drop the is instance check, but that would add another layer of nesting...

Reply:
Fair that makes sense. 
I kinda like the else because we remove all of the unnecessary `isinstance` checks but I don't mind.

Reply:
Changed it